cronjobs/add_authority_external_ref._automation.py

```diff
@@ -0,0 +1,198 @@
+#!/bin/python3
+#
+#    @brief  The automation script to organise process of checking the new release of gnd file
+#               if there is a new release then download the file, extract it, parse using jq, and
+#               generate a new gnd_wiki file using add_autorithy_external_ref.
+#           The output file format is csv.
+#
+#    @author Steven Lolong (steven.lolong@uni-tuebingen.de)
+#
+#    Copyright (C) 2022 Library of the University of Tübingen
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Process logic:
+1. Check whether the gnd file "authorities-gnd-person_lds_*.jsonld.gz" on https://data.dnb.de/opendata/ is newer
+    than the last successful parse to gnd_wiki.csv. 
+    To doing this, there is a config file "/mnt/ZE020150/FID-Entwicklung/ub_tools/config_file_add_authority_ext_ref.cnf" that contain 2 lines of information the first line for the lastest successful date generating gnd_wiki data and the second line is the information about the lastest version on the web.
+    If the date of successful generating gnd_wiki (on first line) is older compare with the one on the web then:
+    a. Download the newer file from the web and put it into folder "/tmp"
+    b. The download file is a zip file, it needs to be extracted first
+    c. Get the information needed by gnd_wiki and put it into the file "/tmp/input_file_for_add_authority_external_ref.txt" using jq:
+        jq -c --stream '.' < authorities-gnd-person_lds.jsonld |grep -E 'https\:/\/d-nb\.info\/gnd\/|wikidata|wikipedia' > input_file_for_add_authority_external_ref.txt
+    d. Run "add_authority_external_ref" program. This program will check whether the lastest version date is newer compare to the lastest successful date generating gnd_wiki, if so then generate a new gnd_wiki.csv file. If it success generating gnd_wiki file then update the date on the config file.
+    e. Update the information in the config file
+
+"""
+
+import os
+import sys
+import re
+import requests
+import util
+from urllib.request import urlopen
+import traceback
+from tqdm import tqdm
+import functools
+import shutil
+
+# this script needs:
+# 1. requests
+# 2. tqdm
+
+# Start global variable
+
+base_url = "https://data.dnb.de/opendata/"
+config_file = "/mnt/ZE020150/FID-Entwicklung/ub_tools/config_file_add_authority_ext_ref.cnf"
```

Comment:
Please note that .cnf (=> .conf) files usually have a special meaning, but this file represents a state rather than a configuration. Also, it seems to be redundant to start a filename with the prefix `config_file_...` and add the extension `.cnf` to it. Also the naming in the middle is not consistant => if the program is called `add_authority_external_ref`, we should name the file exactly the same (and not e.g. use an abbreviation like `ext` for `external`).

Maybe we should call it something like `add_authority_external_ref.state`, `add_authority_external_ref.timestamp` or something similar?